### DISCUSSION TITLE
Route Prefix Doesn't work

### DISCUSSION THREAD

**Discussion by ousid** [livewire/livewire#5576] [3. Feature ideas]
I have been working with Livewire and [Tenancy for Laravel package](https://tenancyforlaravel.com/), using `Stancl\Tenancy\Middleware\InitializeTenancyByPath` Middleware Needs to prefix the routes with `{tenant}` parameter.

### Steps to produce:
- Install `livewire`
- Install & configure `tenancyforlaravel` package
- prefix the routes with `{tenant}` parameter
- create a livewire component
- `RouteIsMissingTenantParameterException` exception occur

I tried every possible way to fix this issue (even altering the livewire route) but no luck.

More info, see here: https://github.com/archtechx/tenancy/issues/628

Is there any possible solution? 

cc @joshhanley

**Comment by joshhanley** [comment]
@ousid Livewire doesn't allow custom prefixes at this stage, so I wouldn't really call it a bug. I've changed it to a feature request 🙂

In regards to the feature, it's not as simple as just adding a prefix due to how Livewire's routes are registered. It is something we are investigating for V3 though.

I noticed that there is a `InitializeTenancyByRequestData` middleware. Maybe you can use that instead for Livewire routes.
https://github.com/archtechx/tenancy/blob/3.x/src/Middleware/InitializeTenancyByRequestData.php

**Reply by ousid** [reply]
Thanks for the answer @joshhanley 

I solved the issue (for now) by extending the `livewire route` & `HttpConnectionHandler` controller class, I know it's not the best approach, but I'll have a second look at `InitializeTenancyByRequestData` middleware at some point.

I hope adding a __prefix__ in livewire routes will be considered in the next version, it will be so much easier to work with such package, or dealing with other logic.

Thanks again Josh!

**Reply by bitlab-infra** [reply]
Hi @ousid , 
I am encountering the same issue. How did you solve the issue by extending the livewire route & HttpConnectionHandler?

**Reply by ousid** [reply]
My solution, not that ideal, and you may face issues with the `livewire` package in the future, in the term of uploading files, and other livewire features.

As @joshhanley said, you may consider using `InitializeTenancyByDomain` or `InitializeTenancyByRequestData`, cause in my case I cannot use both of them.

If you're still interested in the solution, I can send you a gist with the code.

**Reply by bitlab-infra** [reply]
Hi @ousid . Unfortunately, I am not able to use InitializeTenancyByDomain and InitializeTenancyByRequestData as well. I am still interested to see your solution if you can share it and thanks for the head's up on the potential issues it may cause.

**Reply by ousid** [reply]
@bitlab-infra 

See this comment https://github.com/archtechx/tenancy/issues/628#issuecomment-1161863700, it may work for you, and if not, let me know.
